### DISCUSSION TITLE
Passing data from one experiment to another

### DISCUSSION THREAD

**Discussion by yuvalharr** [jspsych/jsPsych#1068] [Q&A] [answered]
Hey there,

I guess this is not strictly a jsPsych question, but I couldn't figure out what I could possibly do. Really hope it's ok :)

For a reason I won't start to explain, I need my experiment to be divided between two different jsPsych experiments. Right after participants finish the first experiment, they are asked to press on a link directing them to the second.

Is there any way you know of that I could take data (e.g. participant ID) from the first experiment and pass it through so it would be saved in the second experiment data as well?

Sorry if this question doesn't have anything to do with jsPsych...

Thanks for any help!

**Comment by becky-gilbert** [comment, marked as answer]
Yes. The trick here is to use URL query parameters to pass information from one page to another by just adding on a key=value pair to the end of the link, after a ?, e.g. 
https://example.com?id=123
If you want to add multiple pieces of info you can add them on separated by a &, e.g.
https://example.com?id=123&condition=A

So in your first experiment, you'll need to add the ID to then end of the link when redirect to the second one. In your second experiment, you'll need to capture the URL parameter and save the ID. You can do this using [jsPsych's URL variables function](https://www.jspsych.org/core_library/jspsych-data/#jspsychdatageturlvariable). And once you have the ID, you can save it to the data using [jsPsych's add properties function](https://www.jspsych.org/core_library/jspsych-data/#jspsychdataaddproperties).

**Reply by yuvalharr** [reply]
Great, thank you very much!
Works perfectly now :)

**Reply by xf15** [reply]
the link to jsPsych's URL variables function above is wrong, that happens :), here https://www.jspsych.org/core_library/jspsych-data/#jspsychdatageturlvariable

**Reply by becky-gilbert** [reply]
@xf15 Oops! Thanks - just fixed the link in my comment 😃
